src/library-authoring/data/slice.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  showLibrarySidebar: false,
+  sidebarBodyComponent: null,
+  showToast: false,
+  toastMessage: null,
```

Comment:
I think it's better to combine these variables. Instead of `showLibrarySidebar` and `sidebarBodyComponent`, just have `sidebarBodyComponent` and allow it to be `undefined` to indicate that the sidebar is closed. Then you don't have to worry about the combination of `showLibrarySidebar: true, sidebarBodyComponent: null` which would be a bug.

Same with `showToast` and `toastMessage`: I would recommend just having:

```typescript
// The list of toast messages that need to be shown to the user. The first one is currently displayed.
const [toastMessages, setToastMessage] = React.useState<string[]>();
// Add a new toast to the queue. It will be displayed immediately if no other toasts are being displayed.
const showToast = React.useCallback((message: string) => setToastMessage((existingMessages) => [...existingMessages, message]);
// After a few seconds, hide the current toast
const toastTimeout = React.useCallback(() => setToastMessage((existingMessages) => [...existingMessages].slice(1));
```

Reply:
Also, we could just have a global `ToastContext` for the whole Course Authoring MFE; it doesn't have to be specific to library home or taxonomies.

Reply:
Added here: https://github.com/open-craft/frontend-app-course-authoring/pull/38/commits/fcfa8489313f995f7f0b03caaafedc85b2ea9fec#diff-9ecb38a4962562f0e05ebbb4799f5d47e5f800f3bec104503115f0a2811ecccdR5

I have placed as a TODO the logic of having a message queue. There may be no budget left in this task